solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -2835,6 +2836,35 @@ public void publishNodeAsDown(String nodeName) {
     } catch (KeeperException e) {
       log.warn("Could not publish node as down: ", e);
     }
+    markPrsReplicasDown();
+  }
+
+  /**
+   * If there are PRS replicas in this node, they dould be immediately marked as down.
+   * Overseer does not mark them as down
+   */
+  private void markPrsReplicasDown() {
+    Map<String, List<String>> collVsReplicas = new HashMap<>(); // replicas for a given collection in this node
+    for (CoreDescriptor descriptor : getCoreContainer().getCoreDescriptors()) {
+      CloudDescriptor cloudDescriptor = descriptor.getCloudDescriptor();
+      DocCollection c = zkStateReader.getClusterState().getCollection(cloudDescriptor.getCollectionName());
+      if(c.isPerReplicaState()) {
+        collVsReplicas.computeIfAbsent(cloudDescriptor.getCollectionName(), s -> new ArrayList<>())
+            .add(cloudDescriptor.getCoreNodeName());
+      }
+    }
+    if(!collVsReplicas.isEmpty()) {
+      for (Map.Entry<String, List<String>> e : collVsReplicas.entrySet()) {
+        DocCollection c = zkStateReader.getClusterState().getCollection(e.getKey());
+        PerReplicaStatesOps prsOps = PerReplicaStatesOps.downReplicas(e.getValue(),
```

Comment:
downreplica call sets leader as false. We should have behavior similar to non-prs collection?

Reply:
Yes, We should change it